src/loader.js

```diff
@@ -156,23 +229,12 @@ async function loader(content) {
             `%${/** @type {number} */ (character.codePointAt(0)).toString(16)}`
         );
   } else {
-    let query = this.resourceQuery;
-
-    if (parsedQuery) {
-      // Remove query param from the bundle due we need that only for bundle purposes
-      const stringifiedParsedQuery = parsedQuery.toString();
-
-      query =
-        stringifiedParsedQuery.length > 0 ? `?${stringifiedParsedQuery}` : "";
-      parsedQuery.delete("as");
-    }
-
     // Old approach for `file-loader` and other old loaders
-    changeResource(this, isAbsolute, output, query);
+    changeResource(this, isAbsolute, output);
 
     // Change name of assets modules after generator
     if (this._module && !this._module.matchResource) {
-      this._module.matchResource = `${output.filename}${query}`;
+      this._module.matchResource = `${output.filename}${this.resourceQuery}`;
```

Comment:
In this case we keep our arguments in query string (i.e. if you will use `[name][ext][query]`, `[query]` will containt `width`/`height`/`as`), I think a developer don't need it because they only related to build pipeline

Reply:
Yes, that because `query` and `this.resourceQuery` are equal here. Take a closer look at this lines in current master branch: https://github.com/webpack-contrib/image-minimizer-webpack-plugin/blob/c45d9dbccab0a878fc5d41325d1149a63c83edf9/src/loader.js#L159-L168
Line 159 stores original `this.resourceQuery`.
Line 163 stringify `parsedQuery` but `parsedQuery` does not modified here, its will change only on line 167.
This means that `query` will contain an `as` argument and will be the same as the original `this.resourceQuery`.

Reply:
Hm, sounds like a bug, we should fix it, I will loom at this

Reply:
Yes, but if fixed, it could be a breaking change 🤔

Reply:
I found it as a bug, because you don't need query params used in build

Reply:
Ok, I fixed it